Sources/UInt128.swift

```diff
@@ -580,19 +496,14 @@ extension UInt128: IntegerArithmetic {
                 // Depending upon which segments we're looking at, we'll want to
                 // check for overflow conditions and flag them when encountered.
                 switch (lhsSegment, rhsSegment) {
-                case (0, 0...2): // lhsSegment 1 * rhsSegment 1 to 3 shouldn't have a value.
-                    if currentValue > 0 { overflow = true }
-                case (0, 3):     // lhsSegment 1 * rhsSegment 4 should only be 32 bits.
-                    if currentValue >> 32 > 0 { overflow = true }
-                case (1, 0...1): // lhsSegment 2 * rhsSegment 1 or 2 shouldn't have a value.
-                    if currentValue > 0 { overflow = true }
-                case (1, 2):     // lhsSegment 2 * rhsSegment 3 should only be 32 bits.
-                    if currentValue >> 32 > 0 { overflow = true }
-                case (2, 0):     // lhsSegment 3 * rhsSegment 1 shouldn't have a value.
+                case (0, 0...2), // lhsSegment 1 * rhsSegment 1 to 3 shouldn't have a value.
+                     (1, 0...1), // lhsSegment 2 * rhsSegment 1 or 2 shouldn't have a value.
+                     (2, 0):     // lhsSegment 3 * rhsSegment 1 shouldn't have a value.
```

Comment:
Nevermind the previous comment. I looked closer and I see you did properly put the statements with the right cases. That's what happens when I do code reviews while almost sleeping :)

This is a 👍 from me :)